### PR TITLE
chore(docs): Update docs to reflect changing showcase/starter submission workflow

### PR DESCRIPTION
**Pull request by AishaBlake**
## Description

With the launch of the new, unified gatsbyjs.com comes some changes to the showcase and starter submission workflow. We want the showcase pages to be useful to the community members who visit them as well as those that submit contributions. That means reevaluating the submission processes and criteria. To allow more time to complete that work, we're pausing showcase submissions for the time being.

I recognize that this represents a higher barrier to entry for some community members who wish to contribute to the repo. Knowing that, I'd like to encourage folks to get started by looking through the [open issues](https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc), particularly those marked ["good first issue"](https://github.com/gatsbyjs/gatsby/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22). Documentation updates are also always welcome. If you'd like to suggest a docs change, please start the conversation by [opening an issue](https://github.com/gatsbyjs/gatsby/issues/new/choose).

### Documentation

Updates to the documentation include:
- Deleting files (and sections of files) describing outdated showcase processes
  - This includes two files which I would otherwise have only altered but were due to be deleted anyway (see #25773)
- Adding redirects for any deleted files
- Adding notes to relevant pages explaining the current pause in showcase submissions

## Outstanding questions

- @hashimwarren, should we be deleting these pages entirely? I expect that we'll need them again but their content will likely be entirely different. If there's a better way to handle this, please let me know.
- @ashumz, please let me know your thoughts on the wording of the notes. Do we want to go into more detail here? We could potentially add something like the second paragraph in my description above.
- Am I missing any sections or pages that should be removed or changed?

### CODE REVIEW

**Comment by hashimwarren** [comment]
Deleting and redirecting is a good plan, since the new content will detail a completely different workflow.

**Comment by ashumz** [comment]
@AishaBlake this looks good to me. I think the messaging is clear for what we know today, and we will follow on by letting the community know the future of the showcase and starter submissions. The one thing I would add is to explicitly share in the PR that we are also pausing starter contributions from the community as we define next steps, for clarity on why those sections are being removed.

File: docs/contributing/community-contributions.md [hunk 1]
```diff
@@ -16,8 +16,8 @@ We welcome all contributions from you in the community, and would be thrilled to
 
 ...and anything else you can think of. Our showcase and library contribution docs are a great place to start:
 
-- [Submit to Site Showcase](/contributing/site-showcase-submissions)
-- [Submit to Starter Library](/contributing/submit-to-starter-library/)
 - [Submit to Plugin Library](/contributing/submit-to-plugin-library/)
 
+**Note:** Showcase contributions are currently paused while the Gatsby team develops a new workflow for sharing your Gatsby sites with the community.
```
Comment (AishaBlake) [inline docs/contributing/community-contributions.md:21]:
```suggestion
**Note:** Showcase and starter contributions are currently paused while the Gatsby team develops a new workflow for sharing your Gatsby sites with the community.
```

File: docs/contributing/how-to-open-a-pull-request.md [hunk 2]
```diff
@@ -95,10 +95,7 @@ Instructions for making changes to the Gatsby source code, tests, internals, API
 
 ### Starters or Site Showcase
 
-There are specific pages about contributing to various parts of the Gatsby ecosystem:
-
-- [Showcase submissions](/contributing/site-showcase-submissions/)
-- [Starter library](/contributing/submit-to-starter-library/)
+**Note:** Showcase contributions are currently paused while the Gatsby team develops a new workflow for sharing your Gatsby sites with the community.
```
Comment (AishaBlake) [inline docs/contributing/how-to-open-a-pull-request.md:98]:
```suggestion
**Note:** Showcase and starters contributions are currently paused while the Gatsby team develops a new workflow for sharing your Gatsby sites with the community.
```

File: docs/docs/starters.md [hunk 3]
```diff
@@ -83,4 +83,3 @@ Learn [how to make a starter](/docs/creating-a-starter/) in the Gatsby docs. Sta
 Community starters are created and maintained by Gatsby community members.
 
 - Looking for a starter for a particular use case? Browse starters that have been submitted to the [Starter Library](/starters/).
```
Comment (AishaBlake) [inline docs/docs/starters.md:85]:
```suggestion
- Looking for a starter for a particular use case? Browse starters that have been submitted to the [Starter Library](/starters/).

**Note:** Showcase and starter contributions are currently paused while the Gatsby team develops a new workflow for sharing your Gatsby sites with the community.
```

**Comment by dospolov** [comment]
why do you limit the community from contributing? I'd like to share my starter, but I can't

**Comment by devHudi** [comment]
I wonder when I can contribute starter again. Please let me know if you have any related documents.

**Pull request merged**